service/src/test/java/bio/terra/cbas/controllers/TestRunSetsApiController.java

```diff
@@ -292,6 +308,34 @@ void setupFunctionalChecks() throws Exception {
                 "ApiException thrown on purpose for testing purposes."));
     when(cromwellService.submitWorkflow(eq(workflowUrl), eq(workflowInputsMap3), any()))
         .thenReturn(new RunId().runId(cromwellWorkflowId3));
+
+    when(samService.getSamUser()).thenReturn(Optional.of(mockUser));
+  }
+
+  @Test
+  void runSetWithSamException() throws Exception {
```

Comment:
Added a test here that might get tossed out or significantly changed when we actually start doing things with this data but is a good sanity check for exception messaging/codes.